server/redis.py

```diff
@@ -25,25 +28,27 @@ def append_to_stream(key: str, data: Dict):
         logger.error(f'Error while writing stream to Redis: {e}')
 
 
-def save_string(key, value, expire=None):
+def save_string(key: str, value, expire: int = None):
+    if type(value) != str:
+        value = json.dumps(value)
     try:
         r.set(key, value, ex=expire)
     except DataError as e:
         logger.error(e)
+        return REDIS_ERROR
 
 
-async def get_string(key, client, caller='id'):
+def get_string(key: str):
     try:
         data = r.get(key)
-        if data is None:
-            await notify_feedback(
-                client,
-                f'{caller} not found',
-            )
-        return data
     except DataError as e:
         logger.error(e)
-        await notify_error_to_client(
-            client,
-            f'DataError in {caller}, send a str',
-        )
+        return REDIS_ERROR
+
+    try:
+        parsed_data = json.loads(data)
+        return parsed_data
+    except json.decoder.JSONDecodeError:
+        return data
+    except TypeError:
+        return data
```

Comment:
In the case of errors, no logging is made?

Reply:
Added a logger for the case of data = None (TypeError) ... for the case of JSONDecodeError I simply return the data without descrambling, since it means that the data is an ID (string) without JSON format (it was not parsed in the method save_string).